pkg/operator/mon/mon_test.go

```diff
@@ -309,3 +311,46 @@ func TestNodeAffinity(t *testing.T) {
 	assert.Equal(t, nodes[1].Name, cleanNodes[0].Name)
 	assert.Equal(t, nodes[2].Name, cleanNodes[1].Name)
 }
+
+func TestHostNetwork(t *testing.T) {
+	clientset := test.New(3)
+	c := New(&clusterd.Context{Clientset: clientset}, "ns", "", "myversion", k8sutil.Placement{}, false)
+	c.clusterInfo = test.CreateClusterInfo(0)
+
+	nodes, err := c.getAvailableMonNodes()
+	assert.Nil(t, err)
+	assert.Equal(t, 3, len(nodes))
+
+	c.HostNetwork = true
+
+	pod := c.makeMonPod(&monConfig{Name: "mon2"}, nodes[2].Name)
+	assert.NotNil(t, pod)
+
+	assert.Equal(t, true, pod.Spec.HostNetwork)
+	assert.Equal(t, v1.DNSClusterFirstWithHostNet, pod.Spec.DNSPolicy)
+}
+
+func TestHostNetworkSetHostIPForMon(t *testing.T) {
+	clientset := test.New(1)
+	node, err := clientset.CoreV1().Nodes().Get("node0", metav1.GetOptions{})
+	assert.Nil(t, err)
+	assert.NotNil(t, node)
+
+	node.Status = v1.NodeStatus{}
+	node.Status.Addresses = []v1.NodeAddress{
+		{
+			Type:    v1.NodeExternalIP,
+			Address: "1.1.1.1",
+		},
+	}
+
+	c := New(&clusterd.Context{Clientset: clientset}, "ns", "", "myversion", k8sutil.Placement{}, false)
+	c.clusterInfo = test.CreateClusterInfo(0)
+
+	mConfig := &monConfig{
+		Name: "mon1",
+	}
+	c.setHostIPForMon(mConfig, *node)
+
+	assert.Equal(t, node.Status.Addresses[0].Address, mConfig.PublicIP)
```

Comment:
would it be simple for this unit test to add another mon on the same host and see a new port number?

Reply:
I added some tests to confirm the port number increase: https://github.com/galexrt/rook/blob/0b8b70fc1c3c723b7af476611c65fc7427b25b33/pkg/operator/mon/mon_test.go#L336-L423